src/vmware/azext_vmware/custom.py

```diff
@@ -166,3 +166,25 @@ def hcxenterprisesite_show(cmd, client: AVSClient, resource_group_name, private_
 
 def hcxenterprisesite_delete(cmd, client: AVSClient, resource_group_name, private_cloud, name):
     return client.hcx_enterprise_sites.delete(resource_group_name=resource_group_name, private_cloud_name=private_cloud, hcx_enterprise_site_name=name)
+
+
+def datastore_create(cmd, client: AVSClient, resource_group_name, private_cloud, cluster, name, nfs_provider_ip=None, nfs_file_path=None, endpoints=[], lun_name=None):
+    from azext_vmware.vendored_sdks.avs_client.models import Datastore, NetAppVolume, DiskPoolVolume
+    datastore = Datastore()
+    if nfs_provider_ip is not None or nfs_file_path is not None:
+        datastore.net_app_volume = NetAppVolume(nfs_provider_ip=nfs_provider_ip, nfs_file_path=nfs_file_path)
+    if len(endpoints) > 0 or lun_name is not None:
+        datastore.disk_pool_volume = DiskPoolVolume(endpoints=endpoints, lun_name=lun_name)
+    return client.datastores.begin_create(resource_group_name=resource_group_name, private_cloud_name=private_cloud, cluster_name=cluster, datastore_name=name, datastore=datastore)
+
+
+def datastore_list(cmd, client: AVSClient, resource_group_name, private_cloud, cluster):
+    return client.datastores.list(resource_group_name=resource_group_name, private_cloud_name=private_cloud, cluster_name=cluster)

```

Comment:
Does it support listing datastores in all resource groups in a subscription? Most Azure CLI list commands support this usage.